File: webapp/src/lib/storage/provider.ts
export default class StorageProvider {
	async init() {
		throw new Error('init method not implemented');
	}

	async uploadLocalFile(
		filename: string,
		uploadedFile: any,
		contentType: string,
		isPublic = false
	): Promise<any> {
		throw new Error('uploadLocalFile method not implemented');
	}

	async uploadBuffer(
		filename: string,
		content: Buffer,
		contentType: string,
		isPublic = false
	): Promise<any> {
		throw new Error('uploadBuffer method not implemented');
	}

	async cloneFile(sourceFilename: string, destinationFilename: string): Promise<any> {
		throw new Error('cloneFile method not implemented');
	}

	async deleteFile(filename: string, isPublic = false): Promise<any> {
		throw new Error('deleteFile method not implemented');
	}

	getBasePath() {
		throw new Error('getBasePath method not implemented.');
	}
}

File: agent-backend/src/storage/provider.py
class StorageProvider:
    def init(self):
        raise NotImplementedError('init method not implemented')

    def upload_file_buffer(self, buffer, filename, file_folder, is_public=False):
        raise NotImplementedError('upload_file_buffer method not implemented')

    def upload_local_file(self, filename, file_folder, is_public=False):
        raise NotImplementedError('upload_local_file method not implemented')

    def delete_file(self, filename, file_folder, is_public=False):
        raise NotImplementedError('delete_file method not implemented')
    
    def get_signed_url(self, filename, file_folder, is_public=False):
        raise NotImplementedError('download_file method not implemented')
